src/par/README.md

```diff
@@ -15,16 +17,500 @@ TritonPart is an open-source constraints-driven partitioner.  It can be used to
+| `-placement_dimension` | Number of dimensions for canvas, if placement information is provided. The default value is `0`, and the allowed values are integers `[0, MAX_INT]`. |
+| `-hypergraph_file` | Path to hypergraph file. |
+| `-fixed_file` | Path to fixed vertices constraint file. |
+| `-community_file` | Path to `community` attributes file to guide the partitioning process. |
+| `-group_file` | Path to `stay together` attributes file. |
+| `-placement_file` | Placement information file, each line corresponds to a group fixed vertices, community and placement attributes following the [hMETIS](https://course.ece.cmu.edu/~ee760/760docs/hMetisManual.pdf) format. |
+| `-e_wt_factors` | Hyperedge weight factor. |
+| `-v_wt_factors` | Vertex weight factors. |
+| `-placement_wt_factors` | Placement weight factors. |
+| `-thr_coarsen_hyperedge_size_skip` | Threshold for ignoring large hyperedge (default 200, integer). |
+| `-thr_coarsen_vertices` | Number of vertices of coarsest hypergraph (default 10, integer). |
+| `-thr_coarsen_hyperedges` | Number of vertices of coarsest hypergraph (default 50, integer). |
+| `-coarsening_ratio` | Coarsening ratio of two adjacent hypergraphs (default 1.6, float). |
+| `-max_coarsen_iters` | Number of iterations (default 30, integer). |
+| `-adj_diff_ratio` | Minimum difference of two adjacent hypergraphs (default 0.0001, float). |
+| `-min_num_vertices_each_part` | Minimum number of vertices in each partition (default 4, integer). |
+| `-num_initial_solutions` | Number of initial solutions (default 50, integer). |
+| `-num_best_initial_solutions` | Number of top initial solutions to filter out (default 10, integer). |
+| `-refiner_iters` | Refinement iterations (default 10, integer). |
+| `-max_moves` | The allowed moves for each Fiduccia-Mattheyes (FM) algorithm pass or greedy refinement (default 60, integer). |
+| `-early_stop_ratio` | Describes the ratio $e$ where if the $n_{moved vertices} > n_{vertices} * e$, the tool exits the current FM pass. The intention behind this is that most of the gains are achieved by the first few FM moves. (default 0.5, float). |
+| `-total_corking_passes` | Maximum level of traversing the buckets to solve the "corking effect" (default 25, integer). |
+| `-v_cycle_flag` | Disables v-cycle is used to refine partitions (default true, bool). |
+| `-max_num_vcycle` | Maximum number of `vcycles` (default 1, integer). |
+| `-num_coarsen_solutions` | Number of coarsening solutions with different randoms seed (default 3, integer). |
+| `-num_vertices_threshold_ilp` | Describes threshold $t$, the number of vertices used for integer linear programming (ILP) partitioning. if $n_{vertices} > t$, do not use ILP-based partitioning.(default 50, integer). |
+| `-global_net_threshold` | If the net is larger than this, it will be ignored by TritonPart (default 1000, integer). |
+
```

Comment:
```suggestion
| `-num_parts` | Number of partitions. The default value is `2`, and the allowed values are integers `[0, MAX_INT]`. |
| `-balance_constraint` | Allowed imbalance between blocks. The default value is `1.0`, and the allowed values are floats. |
| `-base_balance` | Tcl list of baseline imbalance between partitions. The default value is `{1.0}`, and the allowed values are floats that sum up to `1.0`. |
| `-seed` | Random seed. The default value is `0`, and the allowed values are integers `[-MAX_INT, MAX_INT]`. |
| `-vertex_dimension` | Number of vertices in the hypergraph. The default value is `1`, and the allowed values are integers `[0, MAX_INT]`. |
| `-hyperedge_dimension` | Number of hyperedges in hypergraph. The default value is `1`, and the allowed values are integers `[0, MAX_INT]`. |
| `-placement_dimension` | Number of dimensions for canvas if placement information is provided. The default value is `0`, and the allowed values are integers `[0, MAX_INT]`. |
| `-hypergraph_file` | Path to hypergraph file. |
| `-fixed_file` | Path to fixed vertices constraint file. |
| `-community_file` | Path to `community` attributes file to guide the partitioning process. |
| `-group_file` | Path to `stay together` attributes file. |
| `-placement_file` | Placement information file, each line corresponds to a group fixed vertices, community, and placement attributes following the [hMETIS](https://course.ece.cmu.edu/~ee760/760docs/hMetisManual.pdf) format. |
| `-e_wt_factors` | Hyperedge weight factor. |
| `-v_wt_factors` | Vertex weight factors. |
| `-placement_wt_factors` | Placement weight factors. |
| `-thr_coarsen_hyperedge_size_skip` | Threshold for ignoring large hyperedge (default 200, integer). |
| `-thr_coarsen_vertices` | Number of vertices of coarsest hypergraph (default 10, integer). |
| `-thr_coarsen_hyperedges` | Number of vertices of coarsest hypergraph (default 50, integer). |
| `-coarsening_ratio` | Coarsening ratio of two adjacent hypergraphs (default 1.6, float). |
| `-max_coarsen_iters` | Number of iterations (default 30, integer). |
| `-adj_diff_ratio` | Minimum difference of two adjacent hypergraphs (default 0.0001, float). |
| `-min_num_vertices_each_part` | Minimum number of vertices in each partition (default 4, integer). |
| `-num_initial_solutions` | Number of initial solutions (default 50, integer). |
| `-num_best_initial_solutions` | Number of top initial solutions to filter out (default 10, integer). |
| `-refiner_iters` | Refinement iterations (default 10, integer). |
| `-max_moves` | The allowed moves for each Fiduccia-Mattheyes (FM) algorithm pass or greedy refinement (default 60, integer). |
| `-early_stop_ratio` | Describes the ratio $e$ where if the $n_{moved vertices} > n_{vertices} * e$, the tool exits the current FM pass. The intention behind this is that most of the gains are achieved by the first few FM moves. (default 0.5, float). |
| `-total_corking_passes` | Maximum level of traversing the buckets to solve the "corking effect" (default 25, integer). |
| `-v_cycle_flag` | Disables v-cycle is used to refine partitions (default true, bool). |
| `-max_num_vcycle` | Maximum number of `vcycles` (default 1, integer). |
| `-num_coarsen_solutions` | Number of coarsening solutions with different randoms seed (default 3, integer). |
| `-num_vertices_threshold_ilp` | Describes threshold $t$, the number of vertices used for integer linear programming (ILP) partitioning. if $n_{vertices} > t$, do not use ILP-based partitioning.(default 50, integer). |
| `-global_net_threshold` | If the net is larger than this, it will be ignored by TritonPart (default 1000, integer). |

```